pgmpy/tests/test_readwrite/test_ProbModelXML.py

```diff
@@ -714,3 +860,390 @@ def test_file(self):
+                 'LungCancer': {'States': {'no': {}, 'yes': {}},
+                                'role': 'chance',
+                                'type': 'finiteStates',
+                                'Coordinates': {'y': '152', 'x': '421'},
+                                'AdditionalProperties': {'Title': 'L', 'Relevance': '7.0'}}}
+        edges = {'LungCancer': {'TuberculosisOrCancer': {'directed': 'true'}},
+                 'Smoker': {'LungCancer': {'directed': 'true'},
+                            'Bronchitis': {'directed': 'true'}},
+                 'Dyspnea': {},
+                 'X-ray': {},
+                 'VisitToAsia': {'Tuberculosis': {'directed': 'true'}},
+                 'TuberculosisOrCancer': {'X-ray': {'directed': 'true'},
+                                          'Dyspnea': {'directed': 'true'}},
+                 'Bronchitis': {'Dyspnea': {'directed': 'true'}},
+                 'Tuberculosis': {'TuberculosisOrCancer': {'directed': 'true'}}}
+
+        cpds = [{'Values': np.array([[0.95, 0.05], [0.02, 0.98]]),
+                 'Variables': {'X-ray': ['TuberculosisOrCancer']}},
+                {'Values': np.array([[0.7, 0.3], [0.4,  0.6]]),
+                 'Variables': {'Bronchitis': ['Smoker']}},
+                {'Values':  np.array([[0.9, 0.1,  0.3,  0.7], [0.2,  0.8,  0.1,  0.9]]),
+                 'Variables': {'Dyspnea': ['TuberculosisOrCancer', 'Bronchitis']}},
+                {'Values': np.array([[0.99], [0.01]]),
+                 'Variables': {'VisitToAsia': []}},
+                {'Values': np.array([[0.5], [0.5]]),
+                 'Variables': {'Smoker': []}},
+                {'Values': np.array([[0.99, 0.01], [0.9, 0.1]]),
+                 'Variables': {'LungCancer': ['Smoker']}},
+                {'Values': np.array([[0.99, 0.01], [0.95, 0.05]]),
+                 'Variables': {'Tuberculosis': ['VisitToAsia']}},
+                {'Values': np.array([[1, 0, 0, 1], [0, 1, 0, 1]]),
+                 'Variables': {'TuberculosisOrCancer': ['LungCancer', 'Tuberculosis']}}]
+        self.model = BayesianModel(edges_list)
+        for node in nodes:
+            self.model.node[node] = nodes[node]
+        for edge in edges:
+            self.model.edge[edge] = edges[edge]
+
+        tabular_cpds = []
+        for cpd in cpds:
+            var = list(cpd['Variables'].keys())[0]
+            evidence = cpd['Variables'][var]
+            values = cpd['Values']
+            states = len(nodes[var]['States'])
+            evidence_card = [len(nodes[evidence_var]['States'])
+                             for evidence_var in evidence]
+            tabular_cpds.append(
+                TabularCPD(var, states, values, evidence, evidence_card))
+        self.maxDiff = None
+        self.model.add_cpds(*tabular_cpds)
+
+    def test_get_model_data(self):
+        model_data = get_model_data(self.model)
+        model_data_expected = {'probnet': {'Potentials': [{'Values': '0.95 0.05 0.02 0.98 ',
```

Comment:
Wherever there is a test which depends on so much hardcoded data, it is a good idea to put this data in the `setUp` method or in a separate file in a folder called 'testdata' or something, under the tests folder. Since this is a huge dict, maybe make a json file and read from it.